src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,83 @@
-export const Pagination = () => {};
+import React, { useMemo } from 'react';
+import classNames from 'classnames';
+
+type Props = {
+  total: number,
+  perPage: number,
+  currentPage: number;
+  onPageChange: (pageNum: number) => void;
+};
+
+export const Pagination: React.FC<Props> = ({
+  total, perPage, currentPage, onPageChange,
+}) => {
+  const totalPages = useMemo(
+    () => Math.ceil(total / perPage), [total, perPage],
+  );
+
+  const handlePageChange = (pageNum: number) => {
+    onPageChange(pageNum);
+  };
+
+  const generatePages = () => {
+    const pages = [];
+
+    for (let i = 1; i <= totalPages; i += 1) {
+      const page = (
+        <li
+          key={i}
+          className={classNames('page-item', { active: i === currentPage })}
+        >
+          <a
+            data-cy="pageLink"
+            className="page-link"
+            href={`#${i}`}
+            onClick={() => handlePageChange(i)}
+          >
+            {i}
+          </a>
+        </li>
+      );
+
+      pages.push(page);
+    }
+
+    return pages;
+  };
```

Comment:
move this function outside of a component, because every rerender it will be a new function